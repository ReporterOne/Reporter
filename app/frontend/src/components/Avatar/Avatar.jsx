import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { SVGIcon } from '~/components/common';

import {
  Avatar1, Avatar2, Avatar3, Avatar4, Avatar5, Avatar6, Avatar7, Avatar8,
  Av1, Av2, Av3, Av4, Av5, Av6, Av7, Av8, Av9, Av10, Av11, Av12, Av13, Av14,
  Av15, Av16, Av17, Av18, Av19, Av20, Av21, Av22, Av23, Av24, Av25, Av26, Av27,
  Av28, Av29, Av30, Av31, Av32, Av33, Av34, Av35, Av36, Av37,
} from '~/assets/avatars';


const AvatarContainer = styled.div`
  width: ${({ size = 60 }) => size}px;
  height: ${({ size = 60 }) => size}px;
  position: relative;
`;

const AvatarImage = styled(SVGIcon)`
  width: ${({ size = 60 }) => size}px;
  height: ${({ size = 60 }) => size}px;
  position: relative;
  will-change: transform;
  transform: translateY(0);
  &:not(.manual) {
    transition: transform 0.3s;
  }
`;

const Background = styled.div`
  width: ${({ size = 48 }) => size}px;
  height: ${({ size = 48 }) => size}px;
  margin: ${({ spacing = 6 }) => spacing}px;
  border-radius: 50%;
  mask-image: radial-gradient(white, black);
  background-color: ${({ color = 'white' }) => color};
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
`;

const AvatarImageTopHalf = styled(AvatarImage)`
  clip-path: polygon(0 0, 100% 0, 100% 50%, 0 50%);
  position: absolute;
  top: 0;
  left: 0;
`;

const AvatarImageBottomHalf = styled(AvatarImage)`
  clip-path: polygon(0 50%, 100% 50%, 100% 100%, 0% 100%);
  position: absolute;
  top: -${({ spacing = 6 }) => spacing}px;
  left: -${({ spacing = 6 }) => spacing}px;
`;


const avatarsAvailable = [
  Avatar1, Avatar2, Avatar3, Avatar4, Avatar5, Avatar6, Avatar7, Avatar8,
  Av1, Av2, Av3, Av4, Av5, Av6, Av7, Av8, Av9, Av10, Av11, Av12, Av13, Av14,
  Av15, Av16, Av17, Av18, Av19, Av20, Av21, Av22, Av23, Av24, Av25, Av26, Av27,
  Av28, Av29, Av30, Av31, Av32, Av33, Av34, Av35, Av36, Av37,
];

const sized = {
  normal: {
    spacing: 6,
    size: 60,
    backgroundSize: 48
  },
  small: {
    spacing: 5,
    size: 50,
    backgroundSize: 40
  },
  smaller: {
    spacing: 4,
    size: 40,
    backgroundSize: 32
  },
  big: {
    spacing: 7,
    size: 70,
    backgroundSize: 56
  },
  bigger: {
    spacing: 8,
    size: 80,
    backgroundSize: 64
  }
};
export const Avatar = ({ type = 'normal', kind = 8, background = 'white', appearing = 50, ...props }) => {
  const style = useMemo(() => sized[type], [type]);

  return (
    <AvatarContainer size={style.size}>
      {
        appearing < 70 ?
          <Background size={style.backgroundSize} spacing={style.spacing} color={background}>
            <AvatarImage src={avatarsAvailable[kind]} size={style.size}
              style={{
                transform: `translateY(${(100 - appearing)}%)`,
                position: 'absolute', 
                top: `-${style.spacing}px`, 
                left: `-${style.spacing}px`
              }} />
          </Background>
          : (
            <>
              <Background size={style.backgroundSize} spacing={style.spacing} color={background}>
                <AvatarImageBottomHalf src={avatarsAvailable[kind]} spacing={style.spacing} size={style.size} style={{ transform: `translateY(${(100 - appearing)}%)` }} />
              </Background>
              <AvatarImageTopHalf src={avatarsAvailable[kind]} size={style.size} style={{ transform: `translateY(${(100 - appearing)}%)` }} />
            </>
          )
      }
    </AvatarContainer>
  );
}

Avatar.propTypes = {
  type: PropTypes.number,
};


export default Avatar;
