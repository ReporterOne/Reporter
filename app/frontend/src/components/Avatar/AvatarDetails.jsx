import React from 'react';
import styled from 'styled-components';
import Avatar from './Avatar.jsx';
import { Container } from '~/components/common';

const AvatarWrapper = styled(Container)`
  margin: 0 5px;
  display: inline-flex;
  opacity: ${({faded}) => faded? 0.5 : 1};
  will-change: opacity;
  transition: opacity ${({theme}) => theme.animationsSpeed}s cubic-bezier(0.4, 0, 0.2, 1);
`;

const Name = styled(Container)`
  align-items: center;
  justify-content: center;
`;

const NameFragment = styled.span`
  color: white;
  font-weight: 100;  
  font-size: 0.9rem;
  line-height: 105%;
`;


export const AvatarDetails = ({ name, onClick, isFaded = false, ...props }) => {

  return (
    <AvatarWrapper onClick={onClick} faded={isFaded}>
      <Avatar type="big" {...props} />
      <Name>
        {name.split(" ", 1).concat(name.split(" ").slice(1).join(" ")).map((fragment, index) => <NameFragment key={index}>{fragment}</NameFragment>)}
      </Name>
    </AvatarWrapper>
  )
}

export default AvatarDetails;