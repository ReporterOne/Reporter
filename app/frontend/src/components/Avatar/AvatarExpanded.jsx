import React from 'react';
import styled from 'styled-components';
import Avatar from './Avatar.jsx';
import posed from 'react-pose';

import {Container} from '~/components/common';

const AvatarWrapper = styled(Container)`
  margin: 0 5px;
  display: inline-flex;
  opacity: ${({faded}) => faded ? 0.5 : 1};
  will-change: opacity;
  transition: opacity ${({theme}) => theme.animationsSpeed}s 
              cubic-bezier(0.4, 0, 0.2, 1);
  align-items: center;
`;

const Wrapper = styled.div`
  position: relative;
  margin: auto;
  width: 100%;
`;

const AvatarContainer = styled.div`
  position: absolute;
  left: 0;
  justify-content: center;
  align-items: center;
  display: flex;
  box-sizing: border-box;
`;

const Details = styled.div`
  /* fitting to big avatar! */
  border-top-right-radius: ${({rounded}) => rounded ? 30 : 5}px;
  border-bottom-right-radius: ${({rounded}) => rounded ? 30 : 5}px;
  border-top-left-radius: 30px;
  border-bottom-left-radius: 30px;
  min-width: 25px;
  background-color: white;
  height: 54px;
  margin: 6px 5px 0px 5px;
  padding: 2px 0;
  padding-left: 64px;
  overflow: hidden;
  white-space: ${({inline}) => inline ? 'pre-wrap' : 'nowrap'};
  width: ${({inline}) => inline ? '100px' : 'auto'};
  display: flex;
  flex-direction: column;
`;

const Content = styled(Container)`
  justify-content: center;
`;


const Name = styled(Container)`
  color: gray;
  font-weight: 600;
  padding-right: 10px;
`;

const Reason = styled.div`
  justify-content: center;
  align-items: center;
  font-size: 1.8rem;
  line-height: 100%;
  display: flex;
`;

const PosedWrapper = posed(Wrapper)({
  enter: {
    width: '100%',
    delay: ({innerDelay}) => innerDelay * 100,
  },
  exit: {
    width: '0%',
    delay: ({innerDelay}) => innerDelay * 100,
  },
});


export const AvatarExpanded = (
    {
      name, details, onClick, onAvatarTouchStart, delay = 0,
      rounded = false, inline = false, ...props
    },
) => {
  return (
    <AvatarWrapper onClick={onClick}>
      <PosedWrapper innerDelay={delay}>
        <AvatarContainer onTouchStart={onAvatarTouchStart}>
          <Avatar squared type="big" {...props} />
        </AvatarContainer>
        <Details row rounded={rounded} inline={inline}>
          <Content stretched>
            <Name>{name}</Name>
            <Reason>{details}</Reason>
          </Content>
        </Details>
      </PosedWrapper>
    </AvatarWrapper>
  );
};

export default AvatarExpanded;
