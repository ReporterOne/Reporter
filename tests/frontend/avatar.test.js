import React from 'react';
import {render, fireEvent, act} from '@testing-library/react';

import {Avatar} from '~/components/Avatar/Avatar';
import AvatarDetails from '~/components/Avatar/AvatarDetails';
import AvatarExpanded from '~/components/Avatar/AvatarExpanded';


describe('Test avatar component', () => {
  test('Basic Avatar', () => {
    act(() => {
      const {container, getByRole} = render(<Avatar/>);
      expect(container.firstChild).toMatchSnapshot();
      fireEvent.click(getByRole('img')); // assert it does nothing on click
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  test('Avatar jumping', () => {
    act(() => {
      const {container, getByRole} = render(<Avatar jumping={true}/>);
      fireEvent.click(getByRole('img'));
      jest.runAllTimers();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('Test Avatar Details component', () => {
  test('Basic Avatar Details', () => {
    act(() => {
      const {container} = render(<AvatarDetails name="User Name"/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('Test Avatar Expanded component', () => {
  test('Basic Avatar Expanded', () => {
    act(() => {
      const {container} = render(<AvatarExpanded name="User Name"/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
