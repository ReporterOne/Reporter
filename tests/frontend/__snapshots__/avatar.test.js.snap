// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Avatar Details component Basic Avatar Details 1`] = `
.c1 {
  position: relative;
  display: inline-block;
}

.c4 {
  width: 30px;
  height: 30px;
  width: 62px;
  height: 62px;
  position: absolute;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  bottom: 0;
  left: -2px;
  margin: auto;
  will-change: transform;
}

.c4:not(.avatarManual) {
  -webkit-transition: -webkit-transform s cubic-bezier(0.4,0,0.2,1);
  -webkit-transition: transform s cubic-bezier(0.4,0,0.2,1);
  transition: transform s cubic-bezier(0.4,0,0.2,1);
}

.c3 {
  width: 58px;
  height: 58px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  background-color: white;
  border-top-left-radius: 50%;
  border-top-right-radius: 50%;
  bottom: 0;
  left: 0;
  margin: 0;
}

.c2 {
  width: 58px;
  height: 58px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  border-bottom-left-radius: 50%;
  border-bottom-right-radius: 50%;
  padding-top: 4px;
}

.c5 {
  position: absolute;
  bottom: 0;
  right: 5px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background-color: transperant;
}

.c0 {
  position: relative;
  overflow-y: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  min-width: 0;
  min-height: 0;
  margin: 0 5px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  opacity: 1;
  will-change: opacity;
  -webkit-transition: opacity s cubic-bezier(0.4,0,0.2,1);
  transition: opacity s cubic-bezier(0.4,0,0.2,1);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  position: relative;
  overflow-y: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  min-width: 0;
  min-height: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  color: white;
  font-weight: 100;
  font-size: 0.9rem;
  line-height: 105%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      size="58"
    >
      <div
        class="c3"
        size="58"
      />
      <img
        class="c4 avatarAutomatic"
        size="62"
        src="test-file-stub"
        style="transform: translateY(0%);"
      />
    </div>
    <div
      class="c5"
      offset="5"
    />
  </div>
  <div
    class="c6"
  >
    <span
      class="c7"
    >
      User
    </span>
    <span
      class="c7"
    >
      Name
    </span>
  </div>
</div>
`;

exports[`Test Avatar Expanded component Basic Avatar Expanded 1`] = `
.c3 {
  position: relative;
  display: inline-block;
}

.c6 {
  width: 30px;
  height: 30px;
  width: 62px;
  height: 62px;
  position: absolute;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  bottom: 0;
  left: -2px;
  margin: auto;
  will-change: transform;
}

.c6:not(.avatarManual) {
  -webkit-transition: -webkit-transform s cubic-bezier(0.4,0,0.2,1);
  -webkit-transition: transform s cubic-bezier(0.4,0,0.2,1);
  transition: transform s cubic-bezier(0.4,0,0.2,1);
}

.c5 {
  width: 58px;
  height: 58px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  background-color: white;
  border-top-left-radius: 50%;
  border-top-right-radius: 50%;
  bottom: 0;
  left: 0;
  margin: 0;
}

.c4 {
  width: 58px;
  height: 58px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  border-bottom-left-radius: 50%;
  border-bottom-right-radius: 0;
  padding-top: 4px;
}

.c7 {
  position: absolute;
  bottom: 0;
  right: 5px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background-color: transperant;
}

.c0 {
  position: relative;
  overflow-y: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  min-width: 0;
  min-height: 0;
  margin: 0 5px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  opacity: 1;
  will-change: opacity;
  -webkit-transition: opacity s cubic-bezier(0.4,0,0.2,1);
  transition: opacity s cubic-bezier(0.4,0,0.2,1);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: relative;
  margin: auto;
  width: 100%;
}

.c2 {
  position: absolute;
  left: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
}

.c8 {
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top-left-radius: 30px;
  border-bottom-left-radius: 30px;
  min-width: 25px;
  background-color: white;
  height: 54px;
  margin: 6px 5px 0px 5px;
  padding: 2px 0;
  padding-left: 64px;
  overflow: hidden;
  white-space: nowrap;
  width: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  position: relative;
  overflow-y: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  min-width: 0;
  min-height: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  position: relative;
  overflow-y: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  min-width: 0;
  min-height: 0;
  color: gray;
  font-weight: 600;
  padding-right: 10px;
}

.c11 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.8rem;
  line-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    class="c1"
    style="width: 100%;"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
          size="58"
        >
          <div
            class="c5"
            size="58"
          />
          <img
            class="c6 avatarAutomatic"
            size="62"
            src="test-file-stub"
            style="transform: translateY(0%);"
          />
        </div>
        <div
          class="c7"
          offset="5"
        />
      </div>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          User Name
        </div>
        <div
          class="c11"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Test avatar component Avatar jumping 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c3 {
  width: 30px;
  height: 30px;
  width: 52px;
  height: 52px;
  position: absolute;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  bottom: 0;
  left: -2px;
  margin: auto;
  will-change: transform;
}

.c3:not(.avatarManual) {
  -webkit-transition: -webkit-transform s cubic-bezier(0.4,0,0.2,1);
  -webkit-transition: transform s cubic-bezier(0.4,0,0.2,1);
  transition: transform s cubic-bezier(0.4,0,0.2,1);
}

.c2 {
  width: 48px;
  height: 48px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  background-color: white;
  border-top-left-radius: 50%;
  border-top-right-radius: 50%;
  bottom: 0;
  left: 0;
  margin: 0;
}

.c1 {
  width: 48px;
  height: 48px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  border-bottom-left-radius: 50%;
  border-bottom-right-radius: 50%;
  padding-top: 4px;
}

.c4 {
  position: absolute;
  bottom: 0;
  right: 0px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background-color: transperant;
}

<div
  class="c0"
>
  <div
    class="c1"
    size="48"
  >
    <div
      class="c2"
      size="48"
    />
    <img
      class="c3 avatarAutomatic"
      size="52"
      src="test-file-stub"
      style="transform: translateY(0%);"
    />
  </div>
  <div
    class="c4"
    offset="0"
  />
</div>
`;

exports[`Test avatar component Basic Avatar 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c3 {
  width: 30px;
  height: 30px;
  width: 52px;
  height: 52px;
  position: absolute;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  bottom: 0;
  left: -2px;
  margin: auto;
  will-change: transform;
}

.c3:not(.avatarManual) {
  -webkit-transition: -webkit-transform s cubic-bezier(0.4,0,0.2,1);
  -webkit-transition: transform s cubic-bezier(0.4,0,0.2,1);
  transition: transform s cubic-bezier(0.4,0,0.2,1);
}

.c2 {
  width: 48px;
  height: 48px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  background-color: white;
  border-top-left-radius: 50%;
  border-top-right-radius: 50%;
  bottom: 0;
  left: 0;
  margin: 0;
}

.c1 {
  width: 48px;
  height: 48px;
  margin: 2px;
  -webkit-mask-image: radial-gradient(white,black);
  mask-image: radial-gradient(white,black);
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  border-bottom-left-radius: 50%;
  border-bottom-right-radius: 50%;
  padding-top: 4px;
}

.c4 {
  position: absolute;
  bottom: 0;
  right: 0px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background-color: transperant;
}

<div
  class="c0"
>
  <div
    class="c1"
    size="48"
  >
    <div
      class="c2"
      size="48"
    />
    <img
      class="c3 avatarAutomatic"
      size="52"
      src="test-file-stub"
      style="transform: translateY(0%);"
    />
  </div>
  <div
    class="c4"
    offset="0"
  />
</div>
`;

exports[`Test avatar component Basic Avatar 2`] = `
<div
  class="Avatar__AvatarContainer-sc-10jec1a-0 cRADSB"
>
  <div
    class="Avatar__Background-sc-10jec1a-2 Avatar__BackgroundBottomHalf-sc-10jec1a-4 joExKC"
    size="48"
  >
    <div
      class="Avatar__Background-sc-10jec1a-2 Avatar__BackgroundTopHalf-sc-10jec1a-3 eOlcQZ"
      size="48"
    />
    <img
      class="common__Icon-tzqv5k-0 Avatar__AvatarImage-sc-10jec1a-1 dRIMju avatarAutomatic"
      size="52"
      src="test-file-stub"
      style="transform: translateY(0%);"
    />
  </div>
  <div
    class="Avatar__Status-sc-10jec1a-5 faAWeY"
    offset="0"
  />
</div>
`;
